fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
There's no single base class for those that do, but there's an abstract `Dataset` class from which all datasets inherit. We could implement a static method in there so it's available to all datasets if need be.